gateways/paypalpayflow/request_builder.go

```diff
@@ -0,0 +1,87 @@
+package paypalpayflow
+
+import (
+	"fmt"
+
+	"github.com/BoltApp/sleet"
+)
+
+var (
+	defaultVerbosity    string = "HIGH"
+	defaultTender       string = "C"
+	defaultMIT          string = "MIT"
+	MITUnscheduled      string = "MITR"
+	CITUnscheduled      string = "CITU"
+	CITInitial          string = "CITI"
+	CITInitialRecurring string = "CITR"
+	MITRecurring        string = "MITR"
+)
+
+func buildAuthorizeParams(request *sleet.AuthorizationRequest) *Request {
+	expirationDate := fmt.Sprintf("%02d%02d", request.CreditCard.ExpirationMonth, request.CreditCard.ExpirationYear%100)
+	amount := sleet.AmountToDecimalString(&request.Amount)
+	var CardOnFile *string = nil
+
+	if request.ProcessingInitiator != nil {
+		switch *request.ProcessingInitiator {
+		case sleet.ProcessingInitiatorTypeInitialRecurring:
+			CardOnFile = &CITInitialRecurring
+		case sleet.ProcessingInitiatorTypeFollowingRecurring:
+			CardOnFile = &CITUnscheduled
+		case sleet.ProcessingInitiatorTypeStoredMerchantInitiated:
+			CardOnFile = &MITUnscheduled
+		case sleet.ProcessingInitiatorTypeStoredCardholderInitiated:
+			CardOnFile = &CITUnscheduled
+		case sleet.ProcessingInitiatorTypeInitialCardOnFile:
+			CardOnFile = &CITInitial
+		}
+	}
+
+	return &Request{
```

Comment:
For the initiator flags, if available we will pass in the previous transactionId (for example, a subscription). This leads to higher auth rates.

Going off paypal docs seems like we need another param: TXID

Sleet does not own the logic of when this field should be populated -> we will pass it in here: https://github.com/BoltApp/sleet/blob/a68349a55c9f8b1b25bbae56a752ebd5efbc694e/types.go#L89

Reply:
Paypal does seem to require this field for certain initiator flags, thats okay we will just pass that error back to whoever is using sleet

Reply:
Done